Lib/email/mime/image.py

```diff
@@ -34,119 +34,64 @@ def __init__(self, _imagedata, _subtype=None,
         constructor, which turns them into parameters on the Content-Type
         header.
         """
-        _subtype = _what(_imagedata) if _subtype is None else _subtype
-        if _subtype is None:
+        _subtype = _subtype or _infer_subtype(_imagedata)
+        if not _subtype:
             raise TypeError('Could not guess image MIME subtype')
```

Comment:
Perhaps the TypeError should be raised by the infer subtype method as it's fail case, instead of returning `None`, as in this other PR: https://github.com/python/cpython/pull/102540/files

Reply:
Note to self, remove the `None` from the rtype if this is the way we decide to go.